WordPress/src/main/java/org/wordpress/android/ui/reader/utils/ReaderUtils.java

```diff
@@ -359,33 +359,32 @@ public static boolean isFollowing(
         }
     }
 
-    public static ReaderTag getValidTagForSharedPrefs(
-            ReaderTag tag,
+    public static @NonNull ReaderTag getValidTagForSharedPrefs(
+            @NonNull ReaderTag tag,
             boolean isTopLevelReader,
             FilteredRecyclerView recyclerView
```

Comment:
I don't think this function should know about `recyclerView`, it's just not its job to deal with it and having it here creates an unnecessary coupling.

In this specific case, we can simply pass in the `boolean` that is calculated from `(recyclerView != null && recyclerView.isValidFilter(tag))` to address it.

Reply:
Makes sense! I agree we can bring it out 👍!

Reply:
Similar to my previous comment, I prefer if we can tackle these in the upcoming PRs, but I don't mind making the change if you feel otherwise.